src/lib/mina_net2/mina_net2.ml

```diff
@@ -1027,6 +1040,15 @@ module Helper = struct
         let%bind m = Incoming_stream.of_yojson v |> or_error in
         let stream_idx = m.stream_idx in
         let protocol = m.protocol in
+        Option.iter t.all_peers_seen ~f:(fun all_peers_seen ->
+            let all_peers_seen =
+              Set.add all_peers_seen
```

Comment:
Should we add a size limit? and maybe discard the oldest connections?

Reply:
I don't think so, no. We only care about the number of peers, so capping that seems to somewhat defeat the point of the metric.

Reply:
Should we also dump this set to logs on an interval? Just thinking that could be useful if we want to collect a count of unique peers seen across all of our seeds rather than a single seed.

Reply:
Done, this now logs all the peers every 2 hours. The logs are chunked into batches of 50, hopefully that will stop us from overwhelming stackdriver with a single large log message.